src/lib/cms-content.ts

```diff
@@ -27,74 +43,101 @@ export interface ContentItem {
-  contentType = '',
-  slug = '',
-  shortDescription = '',
-  longDescription = '',
-  externalLink = '',
-  level = '',
-  durationHours = 0,
-  lessons = [],
-  fileDownload = { url: '' },
-}: StrapiItem): ContentItem => ({
-  id,
-  title,
-  contentType: contentType === 'CaseStudy' ? 'Case Study' : contentType,
-  slug,
-  shortDescription,
-  longDescription,
-  externalLink,
-  level,
-  durationHours,
+const itemMap = (item: CMSItem): ContentItem => ({
+  title: item.title,
+  contentType: item.content_type,
+  slug: item.slug,
+  shortDescription: item.short_description,
+  longDescription: item.long_description,
+  externalLink: item.external_link,
+  level: item.level,
+  durationHours: item.duration_hours,
   // The below properties actually have many more fields in the CMS, so we filter them out here.
-  lessons: lessons.map(({ id, link, title }: Lesson) => ({
-    id,
-    title,
-    link,
-  })),
-  fileDownload: fileDownload.url,
+  lessons: item.lessons.map(({ lesson }) => lesson),
+  fileDownload: item.file_downloadConnection.edges.length
+    ? item.file_downloadConnection.edges[0].node.url
+    : '',
 });
 
-const getHeaders = (): { 'strapi-token': string } => {
-  const API_KEY = process.env['CMS_API_KEY'];
-  if (!API_KEY) {
-    throw Error('CMS_API_KEY is not defined.');
+const academiaFields = ` 
+      content_type
+      duration_hours
+      external_link
+      file_downloadConnection {
+        edges {
+          node {
+            url
+          }
+        }
+      }
+      lessons {
+        ... on AcademiaLessonsLesson {
+          __typename
+          lesson {
+            link
+            title
+          }
+        }
+      }
+      level
+      long_description
+      short_description
+      slug
+      spoken_language
+      title
+`;
+
+const getAllAcademiaQuery = gql`
+  query all_academia_homepage {
+    all_academia_homepage {
+      items {
+        additional_educator_resourcesConnection(limit: 50) {
+          edges {
+            node {
+              ... on Academia {
+                ${academiaFields}
+              }
+            }
+          }
+        }
+        curriculum_resourcesConnection(limit: 50) {
+          edges {
+            node {
+              ... on Academia {
+                ${academiaFields}
+              }
+            }
+          }
+        }
+      }
+    }
   }
-  return {
-    'strapi-token': API_KEY,
-  };
-};
+`;
+
+const getAcademiaBySlugQuery = gql`
+  query all_academia_homepage($slug: String!) {
+    all_academia(where: { slug: $slug }) {
+      items {
+        ${academiaFields}
+      }
+    }
+  }
+`;
 
 export const getAllContent = async () => {
-  const headers = getHeaders();
-  const {
-    data: { curriculumResources, additionalEducatorResources },
-  } = await axios.get<StrapiData>(
-    `${process.env['CMS_URL']}/public/academia-homepage`,
-    { headers }
+  const response: any = await client.query({
+    query: getAllAcademiaQuery,
+  });
```

Comment:
Presumably, the academia content is largely static and does not require immediately updated content from contentstack.

If otherwise, I just want to mention that I passed the `fetchPolicy` to distinguish the use case between `no-cache` and `cache-first`

Reply:
Correct, it is very static.